apps/src/music/analytics/AnalyticsReporter.ts

```diff
@@ -149,6 +149,12 @@ export default class AnalyticsReporter {
     });
   }
 
+  onMiniMusicPlayerButtonClicked(properties?: object) {
+    this.trackUIEvent('Mini music player play button clicked', {
+      ...properties,
+    });
+  }
+
```

Comment:
non-blocking notes -

does the `properties` object really need to be destructured? I see that the rest of the file is always sending in a copy. Is amplitude's `track` function know to be destructive to the input and alter the object?

Alternatively, if we do need the destructuring (say the `track` function does damage the object), it looks like it could be moved out of the various event handlers and into `trackUIEvent`.

Changing line 175 in the `trackUIEvent` method from

```
track(eventType, payload).promise)
```
to
```
track(eventType, {...payload}).promise)
```

would be functionally identical, but we only need to duplicate the object once where it actually goes into `track` instead of all of the handlers doing it.

I'm still guessing that we _probably_ don't need to copy it at all and this is just an errant copy and paste that propagated, but keeping it centralized in the `trackUIEvent` method at least tidies it up.

Anyway, it's also veering off the PR purposes, so can definitely be tagged and bagged for future work instead.

Reply:
Thanks Jim! Brendan asked if we could add a report on a user clicking on the 'Open project' button so that I updated code to:
```  onMiniMusicPlayerButtonClicked(buttonName: string, properties: object) {
    this.trackUIEvent('Mini music player button clicked', {
      buttonName,
      ...properties,
    });
  }
```
So I think now it makes sense to use the spread operator. Thanks!